timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -273,15 +304,198 @@ def persist(self) -> "ChronosModel":
+                    logger.debug(
+                        f"ChronosBolt models can only be fine-tuned with a maximum prediction_length of {fine_tune_prediction_length}. "
+                        f"Setting prediction_length to {fine_tune_prediction_length}."
+                    )
+
+            trainer_kwargs = fine_tune_args["trainer_kwargs"]
+            trainer_kwargs["disable_tqdm"] = trainer_kwargs.get("disable_tqdm", (verbosity < 3))
+            output_dir = Path(trainer_kwargs["output_dir"])
+
+            training_args = TrainingArguments(**trainer_kwargs, **extra_trainer_kwargs)
+            tokenizer_train_dataset = ChronosFineTuningDataset(
+                target_df=train_data,
+                target_column=self.target,
+                context_length=context_length,
+                prediction_length=fine_tune_prediction_length,
+                # if tokenizer exists, then the data is returned in the HF-style format accepted by
+                # the original Chronos models otherwise the data is returned in ChronosBolt's format
+                tokenizer=getattr(self.model_pipeline, "tokenizer", None),
+                mode="training",
+            ).shuffle(fine_tune_args["shuffle_buffer_size"])
+            callbacks = [EvaluateAndSaveFinalStepCallback()]
+
+            if time_limit is not None:
+                callbacks.append(TimeLimitCallback(time_limit=fine_tune_time_limit))
+
+            if val_data is not None:
+                # evaluate on a randomly-sampled subset
+                eval_max_items = (
+                    min(val_data.num_items, fine_tune_args["eval_max_items"])
+                    if fine_tune_args["eval_max_items"] is not None
+                    else val_data.num_items
+                )
+
+                if eval_max_items < val_data.num_items:
+                    eval_items = np.random.choice(val_data.item_ids.values, size=eval_max_items, replace=False)
+                    val_data = val_data.loc[eval_items]
+
+                tokenizer_val_dataset = ChronosFineTuningDataset(
+                    target_df=val_data,
+                    target_column=self.target,
+                    context_length=context_length,
+                    prediction_length=fine_tune_prediction_length,
+                    tokenizer=getattr(self.model_pipeline, "tokenizer", None),
+                    mode="validation",
+                )
+
+            trainer = Trainer(
+                model=self.model_pipeline.inner_model,
+                args=training_args,
+                train_dataset=tokenizer_train_dataset,
+                eval_dataset=tokenizer_val_dataset,
+                callbacks=callbacks,
+            )
+
+            if verbosity < 3:
+                # remove PrinterCallback from callbacks which logs to the console via a print() call,
+                # so cannot be handled by setting the log level
+                trainer.pop_callback(PrinterCallback)
```

Comment:
does this mean that if verbosity > 3 we will print TF stuff to stdout? if you feel like it, it might be worth overriding the `PrinterCallback` with `ChronosLogCallback(PrinterCallback)` that logs to our logger defined above.

Reply:
Added `LoggerCallback`.